pkg/segment/aggregations/streamstats.go

```diff
@@ -332,181 +358,255 @@ func getListElementAsFloatFromWindow(listElement *list.Element) (float64, error)
 	if !correctType {
 		return 0, fmt.Errorf("getListElementAsFloatFromWindow: Error: element in the window is not a RunningStreamStatsWindowElement, it's of type: %T", listElement.Value)
 	}
-	floatVal, err := dtypeutils.ConvertToFloat(windowElement.Value, 64)
-	if err != nil {
-		return 0.0, fmt.Errorf("getListElementAsFloatFromWindow: Error: element in window does not have a numeric value, has value %v, err: %v", windowElement.Value, err)
+	if windowElement.Value.Dtype != utils.SS_DT_FLOAT {
+		return 0.0, fmt.Errorf("getListElementAsFloatFromWindow: Error: element in window does not have a numeric value, has value %v", windowElement.Value)
 	}
 
-	return floatVal, nil
+	return windowElement.Value.CVal.(float64), nil
 }
 
-func manageMinWindow(window *putils.GobbableList, index int, newValue float64, timestamp uint64) error {
+func getListElementFromWindow(listElement *list.Element) (utils.CValueEnclosure, error) {
+	if listElement == nil {
+		return utils.CValueEnclosure{}, fmt.Errorf("getListElementFromWindow: Error: listElement is nil")
+	}
+
+	windowElement, correctType := listElement.Value.(*structs.RunningStreamStatsWindowElement)
+	if !correctType {
+		return utils.CValueEnclosure{}, fmt.Errorf("getListElementFromWindow: Error: element in the window is not a RunningStreamStatsWindowElement, it's of type: %T", listElement.Value)
+	}
```

Comment:
nit: since you're casting to a pointer, change `RunningStreamStatsWindowElement` to `*RunningStreamStatsWindowElement` in the log message

Reply:
addressed in the latest commit.